### PR TITLE
[dynamo][guards][logs] Generate code_parts for debugging

### PR DESCRIPTION
**Pull request by anijain2305**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #134174
* __->__ #134181
* #134039
* #134016
* #133742

Fixes https://github.com/pytorch/pytorch/issues/132692

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

### CODE REVIEW

File: torch/_dynamo/guards.py [hunk 1]
```diff
@@ -259,6 +259,32 @@ def check_verbose(self, x):
...
+            for verbose_code_part in leaf_guard.verbose_code_parts():
+                code_part = verbose_code_part.split("#")[0].rstrip()
+                code_parts.append(code_part)
+            return code_parts
+
+        def visit(mgr):
+            nonlocal tensor_aliasing_guard_seen
+            for guard in mgr.get_leaf_guards():
+                if isinstance(guard, torch._C._dynamo.guards.NO_TENSOR_ALIASING):  # type: ignore[attr-defined]
+                    if not tensor_aliasing_guard_seen:
```
Comment (youkaichao) [inline torch/_dynamo/guards.py:277]: what is this?

**Review by youkaichao — APPROVED** [review]
LGTM in general, thanks for addressing this problem!

Can we add a test from the [documentation](https://pytorch.org/docs/main/torch.compiler_dynamo_overview.html)? That doc is using `code_parts` attributes.

File: torch/_dynamo/guards.py (hunk 1 above)
Comment (anijain2305) [reply torch/_dynamo/guards.py:277]: Tensor aliasing guard is present at many nodes in the tree. It is identical and for logs purpose, we should print it just once.

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**